cf-agent/verify_users_pam.c

```diff
@@ -903,107 +903,6 @@ static bool SetAccountLocked(const char *puser, const char *hash, bool lock)
 
     return SetAccountLockExpiration(puser, lock);
 }
-
-static bool GroupGetUserMembership (const char *user, StringSet *result)
-{
-    bool ret = true;
-    struct group *group_info;
-
-    FILE *fptr = safe_fopen("/etc/group", "r");
-    if (!fptr)
-    {
-        Log(LOG_LEVEL_ERR, "Could not open '/etc/group': %s", GetErrorStr());
-        return false;
-    }
-
-    while (true)
-    {
-        errno = 0;
-        // Use fgetgrent() instead of getgrent(), to guarantee that the
-        // returned group is a local group, and not for example from LDAP.
-        group_info = fgetgrent(fptr);
-        if (!group_info)
-        {
-            // Documentation among Unices is conflicting on return codes. When there
-            // are no more entries, this happens:
-            // Linux = ENOENT
-            // AIX = ESRCH
-            if (errno && errno != ENOENT && errno != ESRCH)
-            {
-                Log(LOG_LEVEL_ERR, "Error while getting group list. (fgetgrent: '%s')", GetErrorStr());
-                ret = false;
-            }
-            break;
-        }
-        // At least on FreeBSD, gr_mem can be NULL:
-        if (group_info->gr_mem != NULL)
-        {
-            for (int i = 0; group_info->gr_mem[i] != NULL; i++)
-            {
-                if (strcmp(user, group_info->gr_mem[i]) == 0)
-                {
-                    StringSetAdd(result, xstrdup(group_info->gr_name));
-                    break;
-                }
-            }
-        }
-#ifdef __FreeBSD__
-        free(group_info);
-#endif
-    }
-
-    fclose(fptr);
-
-    return ret;
-}
-
-static bool EqualGid(const char *key, const struct group *entry)
-{
-    return (atoi(key) == entry->gr_gid);
-}
-
-static bool EqualGroupName(const char *key, const struct group *entry)
-{
-    return (strcmp(key, entry->gr_name) == 0);
-}
-
-#ifdef __FreeBSD__
-struct group *fgetgrent(FILE *stream)
```

Comment:
this actually looks wrong now to me. I may try a FreeBSD VM, seems like this would fail. Thinking of it, exotics won't cover FreeBSD :( So I will fix this.